jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -205,6 +205,19 @@ var module = module;
         // if the macaroon is valid.
         false
       );
+    },
+
+    /**
+      Lists all the templates for cloud credentials for a specific user.
+
+      @method listTemplates
+      @param callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+    */
+    listTemplates: function(callback) {
+      var url = this.jemURL + '/template';
+      return _makeRequest(this.bakery, url, 'GET', null, callback, true);
```

Comment:
To be clear, I am suggesting the translation to be done by endpoints, not at the _makeRequest level, so different expectations is not an issue. On the other hand, implicit expectations can surely become a problem, sooner or later.